adoc/extensions/sycl_khr_command_graph.adoc

```diff
@@ -252,111 +253,423 @@ a@ Constructs a [code]#depends_on_all_leaves# property instance.
+  [code]#syclDevice# does not report support for this extension.
 
-template<>
-class command_graph<graph_state::executable> {
-public:
-    command_graph() = delete;
-};
+* An [code]#exception# with the [code]#errc::invalid# error code if
+  [code]#syclDevice# is not associated with [code]#syclContext#.
 
-}  // namespace sycl::khr
+.[apititle]#Queue constructor#
+[source, role=synopsis,id=api:khr-command-graph-queue-ctor]
+----
+command_graph(const queue& syclQueue, const property_list& propList = {})
 ----
 
-==== Graph State
+_Constraints:_ [code]#State# is [code]#graph_state::modifiable#.
+
+_Effects:_ Equivalent to [code]#command_graph(syclQueue.get_context(),
+syclQueue.get_device(), propList)#
 
-==== Graph Properties [[graph-properties]]
+[[subsec:khr-graph-command-graph-member-funcs]]
+=== Member functions
 
-===== No-Cycle-Check Property
 
-[source,c++]
+.[apidef]#khr::command_graph::finalize#
+[source, role=synopsis,id=api:khr-command-graph-finalize]
 ----
-namespace sycl::khr::property::graph {
-class no_cycle_check {
-  public:
-    no_cycle_check() = default;
-};
-}
+command_graph<graph_state::executable> finalize(
+      const property_list& propList = {}) const
 ----
+_Constraints:_ [code]#State# is [code]#graph_state::modifiable#.
+
+_Effects:_ Synchronous operation that creates a new [code]#command_graph# in the
+executable state with a fixed topology that can be submitted for execution on
+any queue sharing the device and context associated with the graph.
+It is valid to call this method multiple times to create subsequent executable
+graphs.
+It is also valid to continue to add new nodes to the modifiable graph instance
+after calling this function.
+It is valid to finalize an empty graph instance with no recorded commands.
+Zero or more properties can be provided to this function via an instance of
+[code]#property_list#.
+Valid properties for this function are defined in
+<<subsec:khr-graph-command-graph-properties>>.
 
-===== Assume-Buffer-Outlives-Graph Property [[assume-buffer-outlives-graph-property]]
+_Returns:_ A new [code]#command_graph# object in the executable state.
 
-[source,c++]
+_Throws:_ An [code]#exception# with the [code]#errc::invalid# error code if the
+graph contains a command that is not supported by the device.
+
+.[apititle]#khr::command_graph::begin_recording#
+[source, role=synopsis,id=api:khr-command-graph-begin-recording]
 ----
-namespace sycl::khr::property::graph {
-class assume_buffer_outlives_graph {
-  public:
-    assume_buffer_outlives_graph() = default;
-};
-}
+void begin_recording(queue& recordingQueue,                     (1)
+                       const property_list& propList = {})
+
+void begin_recording(const std::vector<queue>& recordingQueues, (2)
+                       const property_list& propList = {})
+
 ----
+_Constraints:_ [code]#State# is [code]#graph_state::modifiable#.
+
+_Effects (1):_ Synchronously changes the state of [code]#recordingQueue# to the
```

Comment:
I would combine these into a single clause which refers to each overload separately.